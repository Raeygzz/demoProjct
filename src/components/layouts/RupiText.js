import React, { Component } from 'react';
import { Text } from 'react-native';


export default class RupiText extends Component {
  render() {
    return (
      <Text style={[{ fontFamily: "Apercu_Regular", color: "#fff", fontSize: 16 }, this.props.style]} minimumFontScale={16} allowFontScaling={this.props.allowFontScaling} adjustsFontSizeToFit={this.props.adjustsFontSizeToFit}>
        {this.props.text}
      </Text>
    );
  }
}


export class RupiBlackText extends Component {
  render() {
    return (
      <Text style={[{ fontFamily: "Apercu_Regular", color: "#223741", fontSize: 16 }, this.props.style]} minimumFontScale={16} allowFontScaling={this.props.allowFontScaling} adjustsFontSizeToFit={this.props.adjustsFontSizeToFit}>
        {this.props.text}
        {this.props.children}
      </Text>
    );
  }
}
