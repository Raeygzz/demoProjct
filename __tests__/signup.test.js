import React from 'react';
import 'react-native';

import Signup from '../src/screens/signup/signup';
import RupiText from '../src/components/layouts/RupiText';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';


import Enzyme from 'enzyme';
import { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });


const navigation = {
  navigate: jest.fn()
}


describe('<Signup />', () => {

  it('renders correctly using (jest)renderer, (jest)expect & (jest)toMatchSnapshot', () => {
    const tree = renderer.create(<Signup />).toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('renders the RupiText text using (enzyme)shallow, (enzyme)expect, (enzyme)contains & (jest)toEqual', () => {
    const wrapper = shallow(<Signup />);
    expect(wrapper.length).toEqual(1);
    expect(wrapper.contains(<RupiText text="Upload User Image" style={{ color: "yellow" }}></RupiText>)).toEqual(true);
  });


  it('renders TouchableOpacity, RupiText, globalStyle, Text, View, ScrollView, NormalButton & RupiTextInput components using (enzyme)expect, (enzyme)find, (enzyme)exists and (jest)toEqual', () => {
    const wrapper = shallow(<Signup />);
    expect(wrapper.find('ScrollView')).toBeTruthy();
    expect(wrapper.find('ScrollView')).toBeDefined();
    expect(wrapper.find('globalStyle')).toBeTruthy();
    expect(wrapper.find('globalStyle')).toBeDefined();
    expect(wrapper.find('Text').exists()).toEqual(true);
    expect(wrapper.find('View').exists()).toEqual(true);
    expect(wrapper.find('RupiText').exists()).toEqual(true);
    expect(wrapper.find('NormalButton').exists()).toEqual(true);
    expect(wrapper.find('RupiTextInput').exists()).toEqual(true);
    expect(wrapper.find('TouchableOpacity').exists()).toEqual(true);
  });


  it('calls the createAccountSubmit method & test onPress functionalityalong with tests this.props.navigation.navigate using (enzyme)shallow, (enzyme)instance, (enzyme)find, (enzyme)first,props,expect,exists, (React)forceUpdate, (jest)jest.spyOn, (jest)toHaveLength', () => {
    const wrapper = shallow(<Signup navigation={navigation} />);
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'createAccountSubmit');

    instance.forceUpdate();
    wrapper.find('NormalButton').first().props().onItemPressed();

    expect(wrapper.exists());
    expect(spy.mock.calls).toHaveLength(1);
    expect(navigation.navigate).toHaveBeenCalledWith("DashboardBar");
  });


  it('test navigation title using (enzyme)shallow,expect,exists & (jest)toBe,toBeTruthy', () => {
    const wrapper = shallow(<Signup />);
    expect(Signup.navigationOptions.title).toBe('Create a new Account');
    expect(wrapper.find('navigationOptions')).toBeTruthy();
  })

})
