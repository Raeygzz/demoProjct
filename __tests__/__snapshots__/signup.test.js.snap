// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> renders correctly using (jest)renderer, (jest)expect & (jest)toMatchSnapshot 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 27,
        "paddingVertical": 50,
      },
      Object {
        "borderTopLeftRadius": 24,
        "borderTopRightRadius": 24,
      },
      Object {
        "backgroundColor": "#171635",
      },
    ]
  }
>
  <RCTScrollView
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "justifyContent": "flex-end",
                "marginBottom": 20,
                "minHeight": 68,
                "overflow": "hidden",
                "paddingTop": 20,
                "position": "relative",
              },
              undefined,
              Object {
                "paddingBottom": 0,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Apercu_Regular",
                "fontSize": 14,
                "left": 0,
                "lineHeight": 17,
                "opacity": 0.68,
                "position": "absolute",
                "top": 26,
              }
            }
          >
            Email Address
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoFocus={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholderTextColor="rgba(255,255,255,0.7)"
            rejectResponderTermination={true}
            returnKeyType="next"
            selectionColor="#71648E"
            style={
              Array [
                Object {
                  "borderBottomColor": "white",
                  "borderBottomWidth": 1,
                  "color": "white",
                  "fontFamily": "Apercu Medium",
                  "fontSize": 18,
                  "padding": 0,
                },
                null,
                undefined,
                Object {
                  "borderBottomColor": "white",
                  "height": 37,
                  "lineHeight": 12,
                },
                null,
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "justifyContent": "flex-end",
                "marginBottom": 20,
                "minHeight": 68,
                "overflow": "hidden",
                "paddingTop": 20,
                "position": "relative",
              },
              undefined,
              Object {
                "paddingBottom": 0,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Apercu_Regular",
                "fontSize": 14,
                "left": 0,
                "lineHeight": 17,
                "opacity": 0.68,
                "position": "absolute",
                "top": 26,
              }
            }
          >
            Password
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholderTextColor="rgba(255,255,255,0.7)"
            rejectResponderTermination={true}
            returnKeyType="next"
            secureTextEntry={true}
            selectionColor="#71648E"
            style={
              Array [
                Object {
                  "borderBottomColor": "white",
                  "borderBottomWidth": 1,
                  "color": "white",
                  "fontFamily": "Apercu Medium",
                  "fontSize": 18,
                  "padding": 0,
                },
                null,
                undefined,
                Object {
                  "borderBottomColor": "white",
                  "height": 37,
                  "lineHeight": 12,
                },
                null,
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "justifyContent": "flex-end",
                "marginBottom": 20,
                "minHeight": 68,
                "overflow": "hidden",
                "paddingTop": 20,
                "position": "relative",
              },
              undefined,
              Object {
                "paddingBottom": 0,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Apercu_Regular",
                "fontSize": 14,
                "left": 0,
                "lineHeight": 17,
                "opacity": 0.68,
                "position": "absolute",
                "top": 26,
              }
            }
          >
            Repeat Password
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={null}
            placeholderTextColor="rgba(255,255,255,0.7)"
            rejectResponderTermination={true}
            returnKeyType="done"
            secureTextEntry={true}
            selectionColor="#71648E"
            style={
              Array [
                Object {
                  "borderBottomColor": "white",
                  "borderBottomWidth": 1,
                  "color": "white",
                  "fontFamily": "Apercu Medium",
                  "fontSize": 18,
                  "padding": 0,
                },
                null,
                undefined,
                Object {
                  "borderBottomColor": "white",
                  "height": 37,
                  "lineHeight": 12,
                },
                null,
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "height": 20,
            }
          }
        >
          <Text>
             
          </Text>
        </View>
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "white",
              "borderRadius": 1000,
              "borderStyle": "dashed",
              "borderWidth": 1,
              "height": 150,
              "justifyContent": "center",
              "opacity": 1,
              "position": "relative",
              "width": 150,
            }
          }
        >
          <Text
            minimumFontScale={16}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Apercu_Regular",
                  "fontSize": 16,
                },
                Object {
                  "color": "yellow",
                },
              ]
            }
          >
            Upload User Image
          </Text>
        </View>
        <View
          style={
            Object {
              "height": 60,
            }
          }
        >
          <Text>
             
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffbcbc",
                "borderColor": "red",
                "borderWidth": 1,
                "height": 100,
                "justifyContent": "center",
                "margin": 10,
                "width": 300,
              },
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#333333",
                "margin": 20,
              }
            }
          >
            TouchableNativeFeedback is not supported on this platform!
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
